### PR TITLE
Some feedback

### PR DESCRIPTION
**Pull request by wkritzinger**
I'll add more comments inline

### CODE REVIEW

File: fasort.sh [hunk 1]
```diff
@@ -2,21 +2,21 @@
 # Author: Noxy (a dhole who likes paws)
 # This script will find and move any FA files in the working directory and move them to artist folders under $SORTDIR.
 
+set -euo pipefail
```
Comment (wkritzinger) [inline fasort.sh:5]: This is always useful, from the bash docs:

https://www.gnu.org/software/bash/manual/html_node/The-Set-Builtin.html

> -e
>   Exit immediately if a pipeline (see Pipelines), which may consist of a single simple command (see Simple 
>   Commands), a list (see Lists), or a compound command (see Compound Commands) returns a non-zero status.

>-u
Treat unset variables and parameters other than the special parameters ‘@’ or ‘*’ as an error when performing parameter expansion. An error message will be written to the standard error, and a non-interactive shell will exit.

>-o pipefail
If set, the return value of a pipeline is the value of the last (rightmost) command to exit with a non-zero status, or zero if all commands in the pipeline exit successfully. This option is disabled by default.

Basically, if you set this, all exit codes for commands must be 0 or you have to do error handling otherwise the script will exit, same for unset variables.

File: fasort.sh [hunk 2]
```diff
@@ -2,21 +2,21 @@
...
+
 # Directory where you want files to end up
 SORTDIR=~/Pictures/FA_Sorted
 
 # Find files in pwd that match what I think is a decent regex for FA image files
-for f in `ls | egrep '^\d+\.[0-9a-zA-Z-]+_.+\.\w+$'`
+find . -iregex '.*/[0-9]+\.[a-z0-9]+_.*' -print0 | while read -d $'\0' f
```
Comment (wkritzinger) [inline fasort.sh:11]: Using find + read is better because: 

* Filenames can contain spaces and even newlines
* No subshell is created and kept running while it is looping through the results
* The `find` command is much more powerful than `ls`

File: fasort.sh [hunk 3]
```diff
@@ -2,21 +2,21 @@
...
 do
 
     # awk out the artist name
-    ARTIST=$(echo $f | awk -F '.' '{print$2}' | awk -F _ '{print$1}')
+    ARTIST="$(printf '%s' "$(basename "$f")" | awk -F '.' '{print$2}' | awk -F _ '{print$1}')"
```
Comment (wkritzinger) [inline fasort.sh:15]: All variables when used as an argument for a command have to be quoted, otherwise spaces in the string will be treated as separate arguments to the command, for example this code:

```
var="a b"
mkdir $var
```
would create two directories called `a` and `b` and this:

```
var="a b"
mkdir "$var"
```
would create a directory called `a b` (with a space)

also using `echo` to pipe a variable to a command is problematic because `echo` can take some switches which could be present in the variable you're trying to `echo`, for example, this

```
var="-n"
echo "$var"
```
would not echo anything, because `-n` is a switch for the `echo` command. `printf '%s' "$var"` is the safest option

File: fasort.sh [hunk 4]
```diff
@@ -2,21 +2,21 @@
...
 
     # Create the directory for the artist name if it doesn't exist already
-    if [ ! -d $SORTDIR/$ARTIST ]; then
-        mkdir -p $SORTDIR/$ARTIST
-    fi
+    mkdir -p "$SORTDIR/$ARTIST"
```
Comment (wkritzinger) [inline fasort.sh:18]: `mkdir -p` is already a no-op when the directory already exists, no need to check it beforehand

**Comment by noxypaws** [comment]
Thank you so much for the feedback and suggestions! This is a great learning opportunity for me. <3
